### PR TITLE
Feature lwdev 4962 add pending orders to used margin

### PR DESCRIPTION
**Pull request by GSailen**

### CODE REVIEW

File: src/MarginTrading.Backend.Services/Migrations/PendingMarginInstrumentMigration.cs [hunk 1]
```diff
@@ -2,42 +2,56 @@
...
+                if (!orders.Any())
+                    return Task.CompletedTask;
+                
+                foreach (var order in orders)
+                {
+                    HandleOrder(order);
+                }
+                
+                //reinit orders cache with modified data
+                _orderCache.InitOrders(orders);
```
Comment (andrey-anokhin) [inline src/MarginTrading.Backend.Services/Migrations/PendingMarginInstrumentMigration.cs]: Wrong list of orders for init..

File: src/MarginTrading.Backend.Services/Caches/OrderCache.cs [hunk 2]
```diff
@@ -50,9 +50,9 @@ public ImmutableArray<Order> GetPending()
             return WaitingForExecutionOrders.GetAllOrders().ToImmutableArray();
         }
 
-        public ImmutableArray<Order> GetPendingForMarginRecalc(string instrument)
+        public ImmutableArray<Order> GetPendingForMarginRecalc(string instrument, OrderStatus? orderStatus = null)
         {
-            return WaitingForExecutionOrders.GetPendingOrdersByMarginInstrument(instrument).ToImmutableArray();
+            return WaitingForExecutionOrders.GetOrdersByMarginInstrument(instrument, orderStatus).ToImmutableArray();
```
Comment (andrey-anokhin) [inline src/MarginTrading.Backend.Services/Caches/OrderCache.cs]: What is the idea? Additional validation?

File: src/MarginTrading.Backend.Services/MigrationService.cs [hunk 3]
```diff
@@ -3,29 +3,57 @@
...
+            foreach (var migration in _migrations)
             {
-                var instance = (AbstractMigration)Activator.CreateInstance(migrationType, null);
-                await instance.Invoke();
+                var migrationName = migration.GetType().Name;
+
+                try
+                {
+                    if (!migrationVersions.TryGetValue(migrationName, out var version)
+                        && version < migration.Version)
```
[the remark this answers is not in the document]
Comment (GSailen) [reply src/MarginTrading.Backend.Services/MigrationService.cs]: Fixed

File: src/MarginTrading.Backend.Services/MigrationService.cs [hunk 4]
```diff
@@ -3,29 +3,57 @@
...
+            _migrations = migrations;
+            _marginTradingBlobRepository = marginTradingBlobRepository;
+            _log = log;
         }
         
         public async Task InvokeAll()
         {
-            foreach (var migrationType in _migrationTypes)
+            var migrationVersions = await _marginTradingBlobRepository.ReadAsync<Dictionary<string, int>>(
+                                     LykkeConstants.StateBlobContainer, "versions") ?? new Dictionary<string, int>();
```
[the remark this answers is not in the document]
Comment (GSailen) [reply src/MarginTrading.Backend.Services/MigrationService.cs]: Not an issue

File: src/MarginTrading.Backend.Services/MigrationService.cs [hunk 5]
```diff
@@ -3,29 +3,57 @@
...
+
+                try
+                {
+                    if (!migrationVersions.TryGetValue(migrationName, out var version)
+                        && version < migration.Version)
+                    {
+                        await migration.Invoke();
+
+                        migrationVersions.Remove(migrationName);
+                        migrationVersions.Add(migrationName, migration.Version);
```
[the remark this answers is not in the document]
Comment (GSailen) [reply src/MarginTrading.Backend.Services/MigrationService.cs]: Not an issue

**Pull request merged**